furniture_bench/envs/furniture_sim_env.py

```diff
@@ -1136,15 +1199,59 @@ def _reset_parts(self, env_idx, parts_poses=None):
                 device=self.device,
             )
 
-        idxs = torch.tensor(
-            list(self.parts_handles.values()), dtype=torch.int32, device=self.device
+        part_actor_idxs = torch.tensor(self.part_actor_idx_by_env[env_idx], device=self.device, dtype=torch.int32)
+        self.isaac_gym.get_sim_actor_count(self.sim)
+        self.isaac_gym.set_actor_root_state_tensor_indexed(
+            self.sim,
+            gymtorch.unwrap_tensor(self.root_tensor),
+            gymtorch.unwrap_tensor(part_actor_idxs),
+            len(part_actor_idxs),
         )
+
+    def _reset_parts_all(self, parts_poses=None):
+        """Resets ALL furniture parts to the initial pose.
+
+        Args:
+            env_idx (int): The index of the environment.
+            parts_poses (np.ndarray): The poses of the parts. If None, the parts will be reset to the initial pose.
+        """
+        for env_idx in range(self.num_envs):
+            for part_idx, part in enumerate(self.furnitures[env_idx].parts):
```

Comment:
How about calling `self._reset_parts(env_idx, parts_poses)` here? I think this will make the code simpler.